api/users/views.py

```diff
@@ -18,11 +21,16 @@ class UserMixin(object):
 
     def get_user(self, check_permissions=True):
         key = self.kwargs[self.node_lookup_url_kwarg]
+        current_user = self.request.user
 
         if key == 'me':
-            return self.request.user
+            if isinstance(current_user, AnonymousUser):
+                raise PermissionDenied
```

Comment:
This is still incorrect. Note that `PermissionDenied` corresponds to a `403` response: https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/exceptions.py#L97 . As per our discussion, we want 401.